ecl/hql/hqlgram2.cpp

```diff
@@ -3333,6 +3352,79 @@ IHqlExpression *HqlGram::lookupSymbol(IHqlScope * scope, IIdAtom * searchName)
     return scope->lookupSymbol(searchName, LSFpublic, lookupCtx);
 }
 
+const char * HqlGram::composeSymbolSuggestionMsg(IHqlExpression * symbol, StringBuffer & msg)
+{
+    if (symbol)
+        msg.append(" - did you perhaps mean \"").append(symbol->queryName()->queryStr()).append("\"?");
+    return msg.str();
+}
+
+IHqlExpression * HqlGram::findNearestSymbol(unsigned & editDistance, const HqlExprArray & symbolArray, const unsigned * distanceArray, unsigned penalty)
+{
+    editDistance = -1;
+    if (symbolArray.empty() || !distanceArray)
+        return NULL;
+
+    if (distanceArray[0] == 0)
+    {
+        editDistance = 0;
+        return LINK(&symbolArray.item(0));
+    }
+
+    IHqlExpression * nearest = NULL;
+    unsigned minDistance = -1;
+    ForEachItemIn (i, symbolArray)
+    {
+
+        unsigned tempDistance = distanceArray[i];
+        if (tempDistance == 0)
+        {
+            editDistance = 0;
+            return LINK(&symbolArray.item(i));
+        }
+
+        if (tempDistance != (unsigned(-1)))
+            tempDistance =+ penalty;
+
+        if (tempDistance < minDistance)
+        {
+            minDistance = tempDistance;
+            editDistance = distanceArray[i];
+            nearest = &symbolArray.item(i);
+        }
+    }
+    return LINK(nearest);
+}
+
+IHqlExpression * HqlGram::findNearestSymbol(unsigned & minDistance, IHqlExpression * s1, bool ok1, unsigned distance1, IHqlExpression * s2, bool ok2, unsigned distance2)
```

Comment:
See general comments.  With a class that kept the best so-far I suspect this is redundant.